### ISSUE TITLE
KOPS 1.16 UPGRADE: Kops Controller stuck in ContainerCreating state. 

### ISSUE THREAD

**Issue by Reinkaos** [kubernetes/kops#8953]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

1.160.

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

Currently running 1.15.7, attempting to upgrade to 1.16.7.

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

I am running a multi master kubernetes cluster on AWS. The issue has occurred when rolling the first master, the kops-controller pod gets stuck in a ContainerCreating state. When describing the pod this is the message:

failed for volume "kops-controller-token-n8ngf" : failed to sync secret cache: timed out waiting for the condition

Kops upgrade cluster
Kops update cluster
Kops rolling-update cluster --instance-group=<master-1> --yes

**5. What happened after the commands executed?**

The rolling update does not validate. 

**6. What did you expect to happen?**

The master running 1.16.7 to be healthy. 

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

**9. Anything else do we need to know?**

**Comment by rifelpet** [comment]
can you paste your cluster spec? googling ` failed to sync secret cache` suggests that this occurs when the cluster doesn't have RBAC enabled, which is now required in 1.16

**Comment by Reinkaos** [comment]
Hey, thanks for your reply. I do have RBAC enabled, here is my cluster spec with sensitive information removed:

```
apiVersion: kops.k8s.io/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: "2018-05-01T02:30:01Z"
  generation: 21
  name: *
spec:
  additionalPolicies:
  api:
    loadBalancer:
      sslCertificate: *
      type: Public
  authentication:
    aws: {}
  authorization:
    rbac: {}
  channel: stable
  cloudProvider: aws
  configBase: *
  encryptionConfig: true
  etcdClusters:
  - etcdMembers:
    - instanceGroup: master-ap-southeast-2a
      name: a
    - instanceGroup: master-ap-southeast-2b
      name: b
    - instanceGroup: master-ap-southeast-2c
      name: c
    name: main
  - etcdMembers:
    - instanceGroup: master-ap-southeast-2a
      name: a
    - instanceGroup: master-ap-southeast-2b
      name: b
    - instanceGroup: master-ap-southeast-2c
      name: c
    name: events
  fileAssets:
  - content: |
      apiVersion: v1
      kind: Pod
      metadata:
        name: aws-encryption-provider
        namespace: kube-system
      spec:
        containers:
        - image: *
          name: aws-encryption-provider
          command:
          - /aws-encryption-provider
          - -key=*
          - -region=ap-southeast-2
          - -listen=/srv/kubernetes/socket.sock
          - -health-port=:8083
          ports:
          - containerPort: 8083
            protocol: TCP
          livenessProbe:
            httpGet:
              path: /healthz
              port: 8083
          volumeMounts:
          - mountPath: /srv/kubernetes
            name: kmsplugin
        hostNetwork: true
        volumes:
        - name: kmsplugin
          hostPath:
            path: /srv/kubernetes
            type: DirectoryOrCreate
        nodeSelector:
          dedicated: master
        tolerations:
        - key: dedicated
          operator: Equal
          value: master
          effect: NoSchedule
    name: aws-encryption-provider.yaml
    path: /etc/kubernetes/manifests/aws-encryption-provider.yaml
    roles:
    - Master
  iam:
    allowContainerRegistry: true
    legacy: false
  kubeAPIServer:
    admissionControl:
    - PodNodeSelector
    - PodSecurityPolicy
    auditLogMaxAge: 10
    auditLogMaxBackups: 1
    auditLogMaxSize: 100
    auditLogPath: /var/log/kube-apiserver-audit.log
    auditPolicyFile: /srv/kubernetes/audit.yaml
  kubeDNS:
    provider: CoreDNS
  kubelet:
    anonymousAuth: false
    authenticationTokenWebhook: true
    authorizationMode: Webhook
    featureGates:
      PodPriority: "true"
  kubernetesApiAccess:
  kubernetesVersion: 1.16.7
  masterInternalName: *
  masterPublicName: *
  networkCIDR: *
  networkID: *
  networking:
    calico: {}
  nonMasqueradeCIDR: *
  sshAccess:
  subnets:
  - cidr: *
    egress: *
    name: ap-southeast-2a
    type: Private
    zone: ap-southeast-2a
  - cidr: *
    egress: *
    name: ap-southeast-2b
    type: Private
    zone: ap-southeast-2b
  - cidr: *
    egress: *
    name: ap-southeast-2c
    type: Private
    zone: ap-southeast-2c
  - cidr: *
    name: utility-ap-southeast-2a
    type: Utility
    zone: ap-southeast-2a
  - cidr: *
    name: utility-ap-southeast-2b
    type: Utility
    zone: ap-southeast-2b
  - cidr: *
    name: utility-ap-southeast-2c
    type: Utility
    zone: ap-southeast-2c
  topology:
    bastion:
      bastionPublicName: *
    dns:
      type: Public
    masters: private
    nodes: private
```

Hope that helps!

**Comment by Reinkaos** [comment]
Another clue I have at the moment is that my encryption provider pod does not get deployed to the new master node. What does the kops-controller pod do?

**Comment by Reinkaos** [comment]
This is potentially related to an issue that I have in the kube api server logs:

```
Flag --experimental-encryption-provider-config has been deprecated, use --encryption-provider-config.
```

**Comment by Reinkaos** [comment]
I have found the issue. It was because on my Pod spec for the static encryption provider pod I was giving the nodeSelector flag which doesn't appear to be required anymore and breaks the scheduling. Without that the pod was able to be scheduled and then the kops controller worked. It looks like this doc has been updated since I last looked https://github.com/kubernetes-sigs/aws-encryption-provider/blob/master/KOPS.md. I'll leave this here incase it helps someone :).
